pkg/cmdlets/auth/login.go

```diff
@@ -185,23 +185,27 @@ func promptToken(credentials *mono_models.Credentials, out output.Outputer, prom
 
 // AuthenticateWithDevice attempts to authenticate this device with the Platform.
 func AuthenticateWithDevice(out output.Outputer) error {
-	deviceCode, err := model.RequestDeviceAuthorization()
+	response, err := model.RequestDeviceAuthorization()
 	if err != nil {
-		return err
+		return locale.WrapError(err, "err_auth_device")
+	}
+	if response.UserCode == nil || response.VerificationURIComplete == nil {
+		logging.Error("Platform API error: device authorization request's response has nil UserCode and/or VerificationURIComplete")
```

Comment:
Anywhere where you return the error you do not need to log it; logging is handled where the error is handled, not where it originates.

Reply:
Okay.